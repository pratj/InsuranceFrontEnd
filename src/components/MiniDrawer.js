import React from 'react';
import clsx from 'clsx';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import List from '@material-ui/core/List';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import HomeIcon from '@material-ui/icons/Home';
import BusinessIcon from '@material-ui/icons/Business';
import LoyaltyIcon from '@material-ui/icons/Loyalty';
import InfoIcon from '@material-ui/icons/Info';
import { useHistory } from "react-router";
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket';
import "./MiniDrawer.css"
import { Button } from '@material-ui/core';

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  footer: {
    width: 800,
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
    whiteSpace: 'nowrap',
  },
  drawerOpen: {
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerClose: {
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: 'hidden',
    width: theme.spacing(7) + 1,
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9) + 1,
    },
  },
  toolbar: {
    display: 'flex',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
}));

export default function MiniDrawer({RenderComponent, location}) {

  const classes = useStyles();
  const theme = useTheme();
  const history = useHistory();

  const [open, setOpen] = React.useState(false);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  const itemsList = [
    {
      text: "Home",
      icon: <HomeIcon/>,
      onClick: () => history.push("/")
    },
    {
      text: "Services",
      icon: <BusinessIcon/>,
    },
    {
      text: "Partners",
      icon: <LoyaltyIcon/>,
      onClick: () => history.push("/partners")
    },
    {
      text: "Products",
      icon: <ShoppingBasketIcon/>,
      onClick: () => history.push("/products")
    },
    {
      text: "About Us",
      icon: <InfoIcon/>,
      onClick: () => history.push("/aboutus")
    }
  ]
  
  return (
    <div className={classes.root} data-test="miniDrawer">
      <CssBaseline />
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open,
        })}
        style={{ background: "gold" }}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            className={clsx(classes.menuButton, {
              [classes.hide]: open,
            })}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h4" noWrap style={{cursor: 'pointer', fontFamily: 'Righteous, cursive',color: 'white'}} onClick={() => history.push("/")}>
            <Button style={{borderRadius:"99999999px", backgroundColor:"#d1a515"}}><img src="favicon.ico" style={{width: "35px", height:"35px" }} alt="not available"/></Button>
            &nbsp;&nbsp;Insurance Bazaar
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        variant="permanent"
        className={clsx(classes.drawer, {
          [classes.drawerOpen]: open,
          [classes.drawerClose]: !open,
        })}
        classes={{
          paper: clsx({
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open,
          }),
        }}
      >
        <div className={classes.toolbar}>
          <IconButton onClick={handleDrawerClose}>
            {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
          </IconButton>
        </div>
        <Divider />
        <List>
          {itemsList.map((item) => {
            const {text, icon, onClick} = item
            return (
              <ListItem button key={text} onClick={onClick}>
                {icon && <ListItemIcon>{icon}</ListItemIcon>}
                <ListItemText primary={text}/>
              </ListItem>
            )
          })}
        </List>
      </Drawer>
      <main className={classes.content}>
        <div className={classes.toolbar} />
        {typeof location.location.state === 'undefined' ? <RenderComponent/> : <RenderComponent locationData={location}/>}
      </main> 
    </div>
  );
}