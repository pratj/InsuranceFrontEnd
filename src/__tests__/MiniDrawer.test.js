import { shallow, mount } from "enzyme";
import MiniDrawer from "../components/MiniDrawer";
import { render, screen } from "@testing-library/react";
import toJson from "enzyme-to-json";
import RenderCard from "../components/RenderCard";

const location = {
  location: {
    state: {
      product: "Car Insurance",
      category: "Motor Insurance",
    },
  },
};

test("renders minidrawer component", () => {
  shallow(<MiniDrawer RenderComponent={RenderCard} location={location} />);
});

test("renders minidrawer component with parent div", () => {
  const wrapper = shallow(
    <MiniDrawer RenderComponent={RenderCard} location={location} />
  );
  expect(wrapper.find('[data-test="miniDrawer"]').length).toBe(1);
});

test("renders Insurance Bazaar heading", () => {
  render(<MiniDrawer RenderComponent={RenderCard} location={location} />);
  expect(screen.getByText(/Insurance Bazaar/i)).toBeInTheDocument();
});

test("accepts miniDrawer props", () => {
  const wrapper = mount(
    <MiniDrawer RenderComponent={RenderCard} location={location} />
  );
  expect(wrapper.props().RenderComponent).toEqual(RenderCard);
  expect(wrapper.props().location).toEqual(location);
});

test("MiniDrawer snapshot", () => {
  const miniDrawerTree = shallow(
    <MiniDrawer RenderComponent={RenderCard} location={location} />
  );
  expect(toJson(miniDrawerTree)).toMatchSnapshot();
});