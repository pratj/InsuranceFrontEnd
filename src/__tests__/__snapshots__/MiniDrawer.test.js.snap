// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniDrawer snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-test="miniDrawer"
>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-3"
    position="fixed"
    style={
      Object {
        "background": "gold",
      }
    }
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-menuButton-5"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <Memo(ForwardRef(MenuIcon)) />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        noWrap={true}
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "cursor": "pointer",
            "fontFamily": "Righteous, cursive",
          }
        }
        variant="h4"
      >
        <WithStyles(ForwardRef(Button))
          style={
            Object {
              "backgroundColor": "#d1a515",
              "borderRadius": "99999999px",
            }
          }
        >
          <img
            alt="not available"
            src="favicon.ico"
            style={
              Object {
                "height": "35px",
                "width": "35px",
              }
            }
          />
        </WithStyles(ForwardRef(Button))>
          Insurance Bazaar
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    className="makeStyles-drawer-7 makeStyles-drawerClose-9"
    classes={
      Object {
        "paper": "makeStyles-drawerClose-9",
      }
    }
    variant="permanent"
  >
    <div
      className="makeStyles-toolbar-10"
    >
      <WithStyles(ForwardRef(IconButton))
        onClick={[Function]}
      >
        <Memo(ForwardRef(ChevronLeftIcon)) />
      </WithStyles(ForwardRef(IconButton))>
    </div>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="Home"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo(ForwardRef(HomeIcon)) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Home"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="Services"
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo(ForwardRef(BusinessIcon)) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Services"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="Partners"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo(ForwardRef(LoyaltyIcon)) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Partners"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="Products"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo(ForwardRef(ShoppingBasketIcon)) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Products"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="About Us"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <Memo(ForwardRef(InfoIcon)) />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="About Us"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(Drawer))>
  <main
    className="makeStyles-content-11"
  >
    <div
      className="makeStyles-toolbar-10"
    />
    <RenderCard
      locationData={
        Object {
          "location": Object {
            "state": Object {
              "category": "Motor Insurance",
              "product": "Car Insurance",
            },
          },
        }
      }
    />
  </main>
</div>
`;
